tests/test_edge_cases.py

```diff
@@ -1933,41 +1975,53 @@ def __init__(self, t1: T1, t2: T2):
             self.t1 = t1
             self.t2 = t2
 
-    match = r'Fields of type(.*)are not supported.'
+    match = (
+        r'Unable to generate pydantic-core schema for (.*)MyGen\[str, bool\](.*). '
+        r'Setting `arbitrary_types_allowed=True` in the model_config may prevent this error.'
+    )
     with pytest.raises(TypeError, match=match):
 
         class Model(BaseModel):
             a: str
             gen: MyGen[str, bool]
 
 
-@pytest.mark.xfail(reason='working on V2')
 def test_hashable_required():
     class Model(BaseModel):
         v: Hashable
 
+        # TODO: Should arbitrary_types_allowed be necessary for Hashable?
+        model_config = dict(arbitrary_types_allowed=True)
```

Comment:
Question 4 ☝️

Reply:
ideally I guess we should have a validator for this.

Reply:
Agreed. We should be able to at least check at build time if `__hash__` and `__eq__` are defined (and maybe that they are not `object.__hash__/__eq__`).